### ISSUE TITLE
JSX transformation issue

### ISSUE THREAD

**Issue by trueadm** [ryansolid/solid#172]
I was playing around with Solid today, and ran into an issue with the given JSX:

```jsx
import { render } from "solid-js/dom";

function Link({
  children,
  href,
  className,
}) {
  return (
    <a href={href || "#"} className={className}>
      {children}
    </a>
  );
}

function Component() {
  return (
    <div>
      <Link>new</Link>
      {" | "}
      <Link>comments</Link>
      {" | "}
      <Link>show</Link>
      {" | "}
      <Link>ask</Link>
      {" | "}
      <Link>jobs</Link>
      {" | "}
      <Link>submit</Link>
    </div>
  );
}

render(() => <Component />, root);
```

The error is `Uncaught TypeError: Cannot read property 'nextSibling' of null`.

I assume it's because your template doesn't take into consideration it needs holes to insert the dynamic components between. One possible solution might be to add comment nodes into the HTML so that you can correctly find the sibling and replace the comment with the new DOM node.

**Comment by ryansolid** [comment]
Thanks for reporting.

I do use the method you mentioned. I just got very aggressive at optimizing those cases out over the years. It's possible I over-optimized and missed a case. I will look into this.

**Comment by ryansolid** [comment]
Keeping this one open the logic is actually too simple as well. I will need to do a bit more work here.

**Comment by ryansolid** [comment]
Ok fixed this one more robustly now with Solid v0.18.7. Now it can detect multiple expressions between 2 strings properly. Don't hesitate to report any other issues you fine. Thanks.
